log/sequencer.go

```diff
@@ -19,17 +19,54 @@ package log
 import (
 	"context"
 	"fmt"
+	"strconv"
+	"sync"
 	"time"
 
 	"github.com/golang/glog"
 	"github.com/google/trillian"
 	"github.com/google/trillian/crypto"
 	"github.com/google/trillian/crypto/sigpb"
 	"github.com/google/trillian/merkle"
+	"github.com/google/trillian/monitoring"
 	"github.com/google/trillian/storage"
 	"github.com/google/trillian/util"
 )
 
+const logIDLabel = "logid"
+
+var (
+	once                   sync.Once
+	seqBatches             monitoring.Counter
+	seqTreeSize            monitoring.Gauge
+	seqLatency             monitoring.Histogram
+	seqDequeueLatency      monitoring.Histogram
+	seqGetRootLatency      monitoring.Histogram
+	seqInitTreeLatency     monitoring.Histogram
+	seqWriteTreeLatency    monitoring.Histogram
+	seqUpdateLeavesLatency monitoring.Histogram
+	seqSetNodesLatency     monitoring.Histogram
+	seqStoreRootLatency    monitoring.Histogram
+	seqCommitLatency       monitoring.Histogram
+)
+
+func createMetrics(mf monitoring.MetricFactory) {
+	if mf == nil {
```

Comment:
This is defensive? Could we require a non nil metrics factory and move the problem up or is that no better?